spec/models/user_spec.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user_mock) { build(:user) }
+  let!(:user_mock2) { create(:user) }
+  let(:updated_user) do
+    create(
```

Comment:
This can just be `user_mock.save` (or `user.save`) and can happen in the relevant specs, not necessary at this level I think. Let me know if I'm wrong or if there are repercussions to doing so.

Reply:
I added a third user because I'm checking once updated it doesn't have the same attributes as before. If I were to use `user`, its own attributes would be changed and I wouldn't be able to check against its original attributes. Let me know if that doesn't make sense?

Reply:
OK, I've actually added an additional user... there is some logic to my madness:
- You need a user to update: `updated user`
- You need to keep the original attributes of the updated user stored somewhere so you can check that the updated user no longer has those attributes after being updated: `user`
- You need a user with the attributes that the updated user will take on stored somewhere so you can check that the updated user's attributes actually changed to them: `user_three`
- `user_two` can't be used with these tests, since it is already stored in the database and will cause problems due to email uniqueness.
- Factory/faker may produce identical emails/passwords for two instances, so you need both a `not_to eq` AND a `to eq` matcher for each.

Reply:
Yes that makes total sense, great stuff!